plugin-server/src/utils/db/db.ts

```diff
@@ -561,6 +592,85 @@ export class DB {
         return null
     }
 
+    private async getGroupProperty(teamId: number, groupIdentifier: GroupIdentifier): Promise<GroupProperties> {
+        const props = await this.redisGet(
+            this.getGroupPropertiesCacheKey(teamId, groupIdentifier.index, groupIdentifier.key),
+            null
+        )
+        return { identifier: groupIdentifier, properties: props ? (props as Properties) : null }
+    }
+
+    private async getGroupPropertiesFromDbAndUpdateCache(
+        teamId: number,
+        groupIdentifiers: GroupIdentifier[]
+    ): Promise<Record<string, string>> {
+        const query_options: string[] = []
+        const args: any[] = [teamId]
+        let index = args.length + 1
+        for (const gi of groupIdentifiers) {
+            this.statsd?.increment(`group_properties_cache.miss`, {
+                team_id: teamId.toString(),
+                group_type_index: gi.index.toString(),
+            })
+            query_options.push(`(group_type_index = $${index} AND group_key = $${index + 1})`)
+            index += 2
+            args.push(gi.index, gi.key)
+        }
+        const result = await this.postgresQuery(
+            'SELECT group_type_index, group_key, group_properties FROM posthog_group WHERE team_id=$1 AND '.concat(
+                query_options.join(' OR ')
+            ),
+            args,
+            'fetchGroupProperties'
```

Comment:
Can we test that we form the query we expect?

Reply:
you mean a separate test for this, as it's covered by the case when we don't have the values cached